homework_solution.md

```diff
@@ -0,0 +1,474 @@
+# Part 1: Missing Station Data
+
+1. _**Write a query that fills out this table. Try your best to pick the correct station name for each ID. You may have to make some manual choices or editing based on the inconsistencies we've found. Do try to pick the correct name for each station ID based on how popular it is in the trip data.**_
+
+   ### Steps
+
+   1. Create an index on `from_station_id` and `to_station_id` on `trips` table to help query on those
+
+      ```
+      CREATE INDEX idx_from_station ON trips USING btree (from_station_id);
+      ```
+
+      ```
+      CREATE INDEX idx_to_station ON trips USING btree (to_station_id);
+      ```
+
+   2. Delete the exist `stations` table and recreate it with serial ids, so we can rely on auto-increment. (We could try and alter the table, but that seems to be an involved process, and our table being currently empty, it's easier to just recreate it)
+
+      ```
+      DROP TABLE stations;
+      ```
+
+      ```
+      CREATE TABLE stations
+      (
+         id SERIAL,
+         name VARCHAR(100)
+      );
+      ```
+
+      The effect of this can be perceived by running `SELECT nextval(pg_get_serial_sequence('stations', 'id'));`.
+      On the old table, that sequence, though present, wouldn't return a value. Now it does.
+
+   3. Get the first name for stations, assuming that as the correct one where there's more than one, combining the `from` and `to` column results
+
+      ```
+      SELECT from_station_id, (ARRAY_AGG(DISTINCT from_station_name))[1]
+      FROM trips
+      WHERE from_station_id IS NOT NULL
+      GROUP BY from_station_id
+      UNION
+      SELECT from_station_id, (ARRAY_AGG(DISTINCT from_station_name))[1]
+      FROM trips
+      WHERE from_station_id IS NOT NULL
+      GROUP BY from_station_id;
```

Comment:
[major]
This looks the same as the first part of the union. What purpose does it serve?

Reply:
Did you mean to make it `to_station_id`/`to_station_name`?

Reply:
Yes  😄 
I just noticed the mistake while answering the previous comment. Fixing it now.